src/Common/src/TypeSystem/Common/FieldDesc.FieldLayout.cs

```diff
@@ -43,7 +43,7 @@ public virtual bool HasGCStaticBase
             get
             {
                 if (!FieldType.IsValueType)
-                    return true;
+                    return !FieldType.IsPointer;
```

Comment:
Filed #1231 for the rename. I'll do that as soon as this is merged. I dislike pull requests that do half a dozen unrelated things because they make reviewing the changes unnecessarily difficult (not just for the reviewer, but also for me).